tests/MongoDB.Driver.Core.Tests/Specifications/connection-string/TestRunner.cs

```diff
@@ -60,6 +58,172 @@ public void RunTestDefinition(JsonDrivenTestCase testCase)
+            }
+
+            var expectedDocumentElements = expectedDocument
+                .Elements
+                .OrderBy(e => e.Name);
+            var sortedExpectedDocument = new BsonDocument(expectedDocumentElements);
+
+            sortedExpectedDocument.Should().Be(bsonDocument);
+        }
+
+        private void AssertEnum<T>(T? @enum, BsonValue expectedValue) where T : struct
+        {
+            @enum.Should().Be(expectedValue.IsBsonNull ? null : (T?)Enum.Parse(typeof(T), expectedValue.AsString, true));
+        }
+
+        private void AssertOptions(ConnectionString connectionString, BsonDocument definition)
+        {
+            if (definition.TryGetValue("options", out var options) && !options.IsBsonNull)
+            {
+                foreach (var expectedOption in options.AsBsonDocument.Elements)
+                {
+                    var lowerName = expectedOption.Name.ToLower();
+                    switch (lowerName)
+                    {
+                        case "appname":
+                            connectionString.ApplicationName.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "authmechanism":
+                            connectionString.AuthMechanism.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "authmechanismproperties":
+                            AssertDictionary(connectionString.AuthMechanismProperties, expectedOption.Value.AsBsonDocument);
+                            break;
+                        case "authsource":
+                            connectionString.AuthSource.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "compressors":
+                            {
+                                var bsonArray = new BsonArray(connectionString.Compressors.Select(c => c.Type.ToString().ToLowerInvariant()));
+                                var expectedBsonArray = expectedOption.Value.AsBsonArray;
+                                var unsupportedCompressors = new List<BsonValue>();
+                                foreach (var bsonCompressorType in expectedBsonArray)
+                                {
+                                    if (Enum.TryParse<CompressorType>(bsonCompressorType.AsString, true, out var compressorType) && !CompressorSource.IsCompressorSupported(compressorType))
+                                    {
+                                        // do not assert unsupported compressors
+                                        unsupportedCompressors.Add(bsonCompressorType);
+                                    }
+                                }
+                                unsupportedCompressors.ForEach(c => expectedBsonArray.Remove(c));
+                                expectedBsonArray.Should().Be(bsonArray);
+                            }
+                            break;
```

Comment:
Since this is a long complicate switch statement lets try and keep every case as simple as possible with a similar amount of logic in each case, moving additional logic to helper methods.

Refactor this case to:

    case "compressors":
        var compressors = new BsonArray(connectionString.Compressors.Select(c => c.Type.ToString().ToLowerInvariant()));
        var expectedCompressors = RemoveUnsupportedCompressors(expectedOption.Value.AsBsonArray);
        compressors.Should().Be(expectedCompressors);
        break;

with helper method:

    private BsonArray RemoveUnsupportedCompressors(BsonArray compressors)
    {
        bool isSupported(BsonValue value) => 
            Enum.TryParse<CompressorType>(value.AsString, ignoreCase: true, out var type) && 
            CompressorSource.IsCompressorSupported(type);
        return new BsonArray(compressors.Where(c => isSupported(c)));
    }